flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorEndpoint.java

```diff
@@ -725,13 +731,28 @@ public WebMonitorEndpoint(
 	@Override
 	public void startInternal() throws Exception {
 		leaderElectionService.start(this);
+		startExecutionGraphCacheCleanupTask();
+
 		if (hasWebUI) {
 			log.info("Web frontend listening at {}.", getRestBaseUrl());
 		}
 	}
 
+	private void startExecutionGraphCacheCleanupTask() {
+		final long cleanupInterval = 2 * restConfiguration.getRefreshInterval();
```

Comment:
Why did you choose this as the cleanup interval? Why not "just" the refresh interval?

Reply:
The idea was that the cleanup does not have to be as aggressive as the refresh interval. Any other multiple of `getRefreshInterval` would have done the same job I guess.